sisl/sparse.py

```diff
@@ -1646,26 +1692,26 @@ def _ufunc_sp_sp(ufunc, a, b, **kwargs):
     elif not isinstance(b, SparseCSR):
         b = SparseCSR.fromsp(b)
 
-    if not (reduce(op_eq, zip(a.shape[:2], b.shape[:2])) and
-            a.shape[2] == 1 or b.shape[2] == 1 or a.shape[2] == b.shape[2]):
-        raise ValueError(f"could not broadcast sparse matrices {a.shape} and {b.shape}")
+    out = SparseCSR.zeros_from_sparsity_union(a, b)
```

Comment:
I think once this is complete you should even be able to do `SparseCSR.union` on all matrices at once by moving this up (making a combined list `(a + b)`, this should double the performance since you remove the double row loop! ;)

Reply:
Hmm.. perhaps not, since the below creation loop requires that they are in `SparseCSR` anyways... :(

Reply:
I changed this a bit to allow using csr_matrix directly (also small optimization anyway). See the new diff.